lua/ide/components/explorer/filenode.lua

```diff
@@ -152,10 +152,25 @@ FileNode.new = function(path, kind, perms, depth, opts)
                     return
                 end
             end
-            if vim.fn.writefile({}, path) == -1 then
-                error("failed to write file " .. path)
-                return
+
+            -- if path ends with '/' create a directory instead
+            if vim.endswith(path, '/') then
+                self.mkdir(name)
+            else
+                local containing_dir = vim.fn.fnamemodify(path, ':p:h')
+                if vim.fn.isdirectory(containing_dir) == 0 then
+                    if vim.fn.mkdir(containing_dir, 'p') == -1 then
```

Comment:
I think this can delegate to self.mkdir too? Which will handle naming conflicts correctly? Basically, dont call vim.fn.mkdir from touch

Reply:
done.